From 8293230a2a0af8b9370d557a13b324f04c072119 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 19 Jun 2012 21:04:13 -0400
Subject: [PATCH 0025/1936] Fixes uninitialized value within ADnote

- This fixes a possible segfault when changing a part between mono and stero
  mode, or for that matter, it is possible to trigger a bug when dealing with
  the distruction of a mono Adnote, which has not had this value previously set
---
 src/Synth/ADnote.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Synth/ADnote.cpp b/src/Synth/ADnote.cpp
index 2bd2454b..cd376f6c 100644
--- a/src/Synth/ADnote.cpp
+++ b/src/Synth/ADnote.cpp
@@ -1792,6 +1792,8 @@ void ADnote::Global::initparameters(const ADnoteGlobalParam &param,
     GlobalFilterL = Filter::generate(param.GlobalFilter);
     if(stereo)
         GlobalFilterR = Filter::generate(param.GlobalFilter);
+    else
+        GlobalFilterR = NULL;
 
     FilterEnvelope = new Envelope(param.FilterEnvelope, basefreq);
     FilterLfo      = new LFO(param.FilterLfo, basefreq);
-- 
2.47.0

